tests/test_start_date.py

```diff
@@ -0,0 +1,152 @@
+        record_count_by_stream_1 = self.run_and_verify_sync(conn_id_1)
+        synced_records_1 = runner.get_records_from_target_output()
+
+        ##########################################################################
+        # Update START DATE Between Syncs
+        ##########################################################################
+
+        LOGGER.info(f"REPLICATION START DATE CHANGE: {self.start_date} ===>>> {self.start_date_2} ")
+
+        self.start_date = self.start_date_2
+
+        ##########################################################################
+        # Second Sync
+        ##########################################################################
+
+        # Create a new connection with the new start_date
+        conn_id_2 = connections.ensure_connection(self, original_properties=False)
+
+        # Run check mode
+        found_catalogs_2 = self.run_and_verify_check_mode(conn_id_2)
+
+        # Table and field selection
+        test_catalogs_2_all_fields = [
+            catalog for catalog in found_catalogs_2 if catalog.get("tap_stream_id") in expected_streams
+        ]
+        self.perform_and_verify_table_and_field_selection(conn_id_2, test_catalogs_2_all_fields, select_all_fields=True)
+
+        # Run sync
+        record_count_by_stream_2 = self.run_and_verify_sync(conn_id_2)
+        synced_records_2 = runner.get_records_from_target_output()
+
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+                # Expected values
+                expected_primary_keys = self.expected_primary_keys()[stream]
+                expected_metadata = self.expected_metadata()[stream]
+                expected_start_date_1 = self.timedelta_formatted(self.start_date_1, self.START_DATE_FORMAT, -1)
+                expected_start_date_2 = self.timedelta_formatted(self.start_date_2, self.START_DATE_FORMAT, -1)
+
+                # Collect information for assertions from syncs 1 & 2 base on expected values
+                record_count_sync_1 = record_count_by_stream_1.get(stream, 0)
+                record_count_sync_2 = record_count_by_stream_2.get(stream, 0)
+
+                primary_keys_list_1 = [
+                    tuple(message.get("data").get(expected_pk) for expected_pk in expected_primary_keys)
+                    for message in synced_records_1.get(stream, {}).get("messages", [])
+                    if message.get("action") == "upsert"
+                ]
+                primary_keys_list_2 = [
+                    tuple(message.get("data").get(expected_pk) for expected_pk in expected_primary_keys)
+                    for message in synced_records_2.get(stream, {}).get("messages", [])
+                    if message.get("action") == "upsert"
+                ]
+
+                primary_keys_sync_1 = set(primary_keys_list_1)
+                primary_keys_sync_2 = set(primary_keys_list_2)
+
+                if expected_metadata.get(self.OBEYS_START_DATE):
```

Comment:
shouldn't this be:
 if expected_metadata[stream].get(self.OBEYS_START_DATE):

Reply:
```python
            with self.subTest(stream=stream):
                # Expected values
                expected_primary_keys = self.expected_primary_keys()[stream]
                expected_metadata = self.expected_metadata()[stream]
```

expected metadata is already fetched for the current stream, hence wont be needed to use the stream name again